code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm

```diff
@@ -0,0 +1,221 @@
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/rust_regen,/datum/eldritch_knowledge/spell/ashen_shift)
+	required_atoms = list(/obj/structure/reagent_dispensers/watertank)
+	result_atoms = list(/obj/item/reagent_containers/glass/beaker/eldritch)
+
+/datum/eldritch_knowledge/rust_final
+	name = "Rustbringer's Oath"
+	desc = "Bring 3 corpses onto the transmutation rune. After you finish the ritual rust will now automatically spread from the rune. Your healing on rust is also tripled, while you become more resillient while on rust tiles."
+	gain_text = "Champion of rust. Corruptor of steel. Fear the dark for Rustbringer has come!"
+	cost = 3
+	required_atoms = list(/mob/living/carbon/human)
+	route = "Rust"
+	var/finished = FALSE
+	///keeps track of previous brute mod
+	var/prev_brute_mod
+	///keeps track of previous burn mod
+	var/prev_burn_mod
+
+/datum/eldritch_knowledge/rust_final/recipe_snowflake_check(list/atoms, loc,list/selected_atoms)
+	if(finished)
+		return FALSE
+	var/counter = 0
+	for(var/mob/living/carbon/human/H in atoms)
+		selected_atoms |= H
+		counter++
+		if(counter == 3)
+			return TRUE
+	return FALSE
+
+/datum/eldritch_knowledge/rust_final/on_finished_recipe(mob/living/user, list/atoms, loc)
+	var/mob/living/carbon/human/H = user
+	prev_brute_mod = H.physiology.brute_mod
+	prev_burn_mod = H.physiology.burn_mod
+	finished = TRUE
+	priority_announce("$^@&#*$^@(#&$(@&#^$&#^@# Fear the decay, for Rustbringer [user.real_name] has come! $^@&#*$^@(#&$(@&#^$&#^@#","#$^@&#*$^@(#&$(@&#^$&#^@#", 'sound/ai/spanomalies.ogg')
+	new /datum/rust_spread(loc)
+
+
+	. = ..()
+
+/datum/eldritch_knowledge/rust_final/on_life(mob/user)
+	. = ..()
+	if(!finished)
+		return
+	var/mob/living/carbon/human/H = user
+	var/turf/T = get_turf(user)
+	if(!istype(T,/turf/open/floor/plating/rust) || !isliving(user))
+		H.physiology.brute_mod = prev_brute_mod
+		H.physiology.burn_mod = prev_burn_mod
+		return
+
+	H.adjustBruteLoss(-3)
+	H.adjustFireLoss(-3)
+	H.adjustToxLoss(-3)
+	H.adjustOxyLoss(-1)
+	H.adjustStaminaLoss(-10)
+	H.physiology.brute_mod = prev_brute_mod * 0.5
+	H.physiology.burn_mod = prev_burn_mod * 0.5
+
+
+/**
+  * #Rust spread datum
+  *
+  * Simple datum that automatically spreads rust around it
+  *
+  * Simple implementation of automatically growing entity
+  */
+/datum/rust_spread
+	var/list/edge_turfs = list()
+	var/list/turfs = list()
+	var/static/list/blacklisted_turfs = typecacheof(list(/turf/closed/wall/rust,/turf/closed/wall/r_wall/rust,/turf/open/space,/turf/open/lava,/turf/open/chasm,/turf/open/floor/plating/rust))
+	var/spread_per_tick = 5
+
+
+/datum/rust_spread/New(loc)
+	var/turf/T = get_turf(loc)
+	T.ChangeTurf(/turf/open/floor/plating/rust)
+	turfs += T
+	START_PROCESSING(SSprocessing,src)
+	. = ..()
+
+/datum/rust_spread/Destroy(force, ...)
+	STOP_PROCESSING(SSprocessing,src)
+	. = ..()
+
+/datum/rust_spread/process()
+	compile_turfs()
+	var/turf/T
+	for(var/i = 0, i < spread_per_tick,i++)
```

Comment:
Use the list loop.

Reply:
god fucking damnit i thought i replaced all of the loops